### ISSUE TITLE
Data Export

### ISSUE THREAD

**Issue by jasonlewis72** [jasonlewis72/amcs#41]
Hi Brad,
Lachlan has asked about exporting the mailing addresses that are collected through the signup process. These some of these are custom fields.

Can they be easily exported?

--------------------------------------------------------------------------------------------------------
Hey Jason, a question from our client. 

Are we collecting mailing addresses in the system? IE: are they easily exportable?

Cheers,

- Lachlan.

**Comment by bradmac** [comment]
Sure, either from the contact requests report..
http://admin.amcs.olhub.com/reporting/contactRequests

Or from the group signups report (effectively deduped)
http://admin.amcs.olhub.com/reporting/groupSignups

**Comment by lachlanrdale** [comment]
Hey guys, not sure this is correct.

Can see no mailing address details in either of those exports.

My question is: is Kademi actively storing this data to begin with? Have we set up the groups such that they sort all contact fields?

**Comment by bradmac** [comment]
Sorry, my bad, i thought you meant email. But the mailing fields we're collecting do appear to be in the contact requests report. Eg:

firstName	surName	email	phone	company	message	optins	Country	Postcode	surName

SUSAN	BUCHAN	susan.buchan1@gmail.com	450645223			Petition-Main-Email,Petition-Main-SMS	Australia	3218	BUCHAN

**Comment by lachlanrdale** [comment]
Donors are added to group “Once-Off-Payment” or “Recurring-Payment”.

No users from these groups appear in the Contact Report.

The specific mailing details I aim referring to are those for our Fundraising page, namely:

* Country
* Address 1
* Address 2
* Suburb/City
* State/Province
* Zip/Post code

**Comment by bradmac** [comment]
right, sorry, i didnt realise you were talking about the donor process. I'll check it out...

**Comment by bradmac** [comment]
OK, the payment process generates a contact request record, and that record does have the correct fields on it. Eg:
http://admin.amcs.olhub.com/manageContacts/41928350

These fields should appear in the contact report, but they dont. The reason is that there is a performance optimisation where only the first 50 records are scanned to get a list of fields. Right now the first 50 records are normal contact requests, so the report isnt outputting the fields from payment form submits.

We're currently porting the contact request report to a new backend framework which will allow improved filtering and performance, and when done that will correctly display all fields collected for the selected criteria.

Kademi issue:
https://github.com/Kademi/kademi-dev/issues/333

**Comment by lachlanrdale** [comment]
Hey @bradmac where did you get to with this?

Our client is after weekly extracts of all new donors, including:

* Name
* Email
* Phone number
* Address
* Donation type
* Donation amount

Please let me know if this is something we'll have to arrange with you directly, or whether it can be downloaded from the Admin panel.

Cheers,

**Comment by bradmac** [comment]
I think the best thing is for us to add a new report for recurring transactions. That will keep them seperate from regular contact requests.

**Comment by lachlanrdale** [comment]
They'd like one-off transaction info too (same detail as above)

**Comment by bradmac** [comment]
OK

**Comment by lachlanrdale** [comment]
Hey @bradmac I note data for this report does not include historical data.

Has this data been saved anywhere in Kademi?

![screen shot 2015-05-13 at 12 17 37 pm](https://cloud.githubusercontent.com/assets/12091538/7602627/17b8f0cc-f96a-11e4-9231-09a00463195f.png)

**Comment by bradmac** [comment]
It should be showing all records in the selected date range. Does that appear to not be the case?

**Comment by lachlanrdale** [comment]
Correct. That report is for "all donations this month." Data is missing.

Cheers,

**Comment by bradmac** [comment]
We've added new columns to the payment transaction record to capture the exact data as it was sent to the payment processor. Payments captured before the new columns were added is captured into the user's profile but could (theoretically) change. We could run a data migration process to copy data from the profile into the blank payment transactions fields. Please let me know if thats needed.

**Comment by lachlanrdale** [comment]
@bradmac That would be perfect, and would cover data to date.

**Comment by skwidge** [comment]
See https://github.com/Kademi/kademi-dev/issues/354

**Comment by lachlanrdale** [comment]
@bradmac Still not seeing this function correctly.

* Graph is showing up, but on table data
* Downloaded CSV, still blank on historical data

![screen shot 2015-05-15 at 4 24 19 pm](https://cloud.githubusercontent.com/assets/12091538/7647865/dd57247e-fb1e-11e4-9c11-1db0ea705111.png)


Could you walk through and shoot me some screenshots of it working in AMCS?

Failing that, a direct link to a working AMCS report is fine too.

**Comment by skwidge** [comment]
@lachlanrdale @bradmac 
I'm afraid that the address data for those transactions was not preserved anywhere in Kademi before we made those changes. I can't get that back. All future transactions will have it though.

**Comment by bradmac** [comment]
Hey @skwidge, could you check the contact requests list? The payment process begins with invoking the normal contactus process, which should record the input fields including address data. It will be linked to the same profile as the transaction

**Comment by skwidge** [comment]
@bradmac @lachlanrdale You were right there was some address info in the contact us tables. It is still not complete, but it has filled a lot of the gaps.

**Comment by lachlanrdale** [comment]
@bradmac @skwidge Are we able to a one-off sync to populate historical data?

Client has asked for this by COB if possible.

Cheers,

**Comment by bradmac** [comment]
We actually did do a one-off migration. The "gaps" should just be test data with incomplete information

**Comment by lachlanrdale** [comment]
Hey @bradmac - I see duplicates abound in this data.

I believe Kademi is incorrectly marking some failed transactions as approved (which is compounding problems I am having with this extract when talking with Bruce).

![screen shot 2015-05-20 at 3 11 23 pm](https://cloud.githubusercontent.com/assets/12091538/7719294/9cc88f0a-ff02-11e4-9451-1a99aa7a546f.png)

**Comment by bradmac** [comment]
OK, I'll go through this with Bruce tomorrow.

**Comment by lachlanrdale** [comment]
Thanks Brad,
